lib/IAMClient.js

```diff
@@ -395,13 +400,33 @@ class IAMClient {
                 ret += receivedData.toString();
             })
 
-            .on('error', callback)
+            .on('error', err => {
+                if (logger) {
+                    logger.debug('error receiving data', {
```

Comment:
Don't we want it at the `error` level instead? That would make sense for an internal error.